lib/cli-engine.js

```diff
@@ -496,15 +517,19 @@ CLIEngine.prototype = {
             results.push(res);
         }
 
-        files.forEach(function(pattern) {
-            glob.sync(pattern, globOptions).forEach(executeOnFile);
+        patterns.forEach(function(pattern) {
+            if (shell.test("-f", pattern) && !ignoredPaths.contains(pattern)) {
```

Comment:
Why not just use `fs` insead of `shell`. And that way we can get rid of the shell dependency and move it back to devDependencies.